marklogic-data-hub-central/src/main/java/com/marklogic/hub/central/entities/search/impl/CreatedOnFacetHandler.java

```diff
@@ -21,33 +21,113 @@
 import com.marklogic.hub.central.entities.search.Constants;
 import com.marklogic.hub.central.entities.search.FacetHandler;
 import com.marklogic.hub.central.entities.search.models.DocSearchQueryInfo;
+import com.marklogic.hub.central.exceptions.DataHubException;
 
 import java.time.LocalDate;
 import java.time.ZoneId;
+import java.time.ZoneOffset;
+import java.time.ZonedDateTime;
 import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.time.temporal.TemporalAdjusters;
+import java.util.HashMap;
+import java.util.Map;
 
 public class CreatedOnFacetHandler implements FacetHandler {
 
-    private static final DateTimeFormatter DATE_FORMAT = DateTimeFormatter.ofPattern("yyyy-MM-dd");
-    private static final DateTimeFormatter DATE_TIME_FORMAT = DateTimeFormatter
-            .ofPattern("yyyy-MM-dd'T'HH:mm:ss.SSSZ");
+    public static final DateTimeFormatter DATE_TIME_FORMAT = DateTimeFormatter.ofPattern("yyyy-MM-dd'T'HH:mm:ssZ");
 
     @Override
     public StructuredQueryDefinition buildQuery(DocSearchQueryInfo.FacetData data, StructuredQueryBuilder queryBuilder) {
-        // Converting the date in string format from yyyy-MM-dd format to yyyy-MM-dd HH:mm:ss format
-        LocalDate startDate = LocalDate.parse(data.getRangeValues().getLowerBound(), DATE_FORMAT);
-        String startDateTime = startDate.atStartOfDay(ZoneId.systemDefault())
-                .format(DATE_TIME_FORMAT);
+        Map<String, String> dateRange = computeDateRange(data, new CreatedOnFacetInputs());
+        return queryBuilder.and(
+            queryBuilder.rangeConstraint(Constants.CREATED_ON_CONSTRAINT_NAME, StructuredQueryBuilder.Operator.GE, dateRange.get("startDateTime")),
+            queryBuilder.rangeConstraint(Constants.CREATED_ON_CONSTRAINT_NAME, StructuredQueryBuilder.Operator.LT, dateRange.get("endDateTime"))
+        );
+    }
+
+    protected Map<String, String> computeDateRange(DocSearchQueryInfo.FacetData data, CreatedOnFacetInputs facetInputs) {
```

Comment:
It's fine to pass CreatedOnFacetInputs in here, but you don't really need to because the caller doesn't care about it. You can just instantiate it within this method. No need to change though.